suite/volume/volume_setup.go

```diff
@@ -0,0 +1,320 @@
+package volume
+
+import (
+	"context"
+	"fmt"
+	"github.com/avast/retry-go"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/smartcontractkit/integrations-framework/client"
+	"github.com/smartcontractkit/integrations-framework/contracts"
+	"github.com/smartcontractkit/integrations-framework/suite"
+	"github.com/smartcontractkit/integrations-framework/tools"
+	"golang.org/x/sync/errgroup"
+	"math/big"
+	"sync"
+	"time"
+)
+
+// FluxTestSpec flux aggregator volume test spec
+type FluxTestSpec struct {
+	TestSpec
+	AggregatorsNum              int
+	RequiredSubmissions         int
+	RestartDelayRounds          int
+	JobPrefix                   string
+	ObservedValueChangeInterval time.Duration
+	NodePollTimePeriod          time.Duration
+	FluxOptions                 contracts.FluxAggregatorOptions
+}
+
+// FluxTest flux test runtime data
+type FluxTest struct {
+	Test
+	// round durations, calculated as a difference from earliest chainlink run for contract,
+	// until all confirmations found on-chain (block_timestamp)
+	roundsDurationData []float64
+	FluxInstances      *[]contracts.FluxAggregator
+	ContractsToJobsMap map[string][]JobByInstance
+	NodesByHostPort    map[string]client.Chainlink
+	AlreadySeenRuns    map[string]bool
+}
+
+// FluxInstanceDeployment data required by flux instance to calculate per round metrics
+type FluxInstanceDeployment struct {
+	InstanceDeployment
+	FluxInstancesMu   *sync.Mutex
+	FluxInstances     *[]contracts.FluxAggregator
+	ContractToJobsMap map[string][]JobByInstance
+	NodesByHostPort   map[string]client.Chainlink
+}
+
+// roundVals structure only for on-chain debug
+type roundVals struct {
+	RoundID int64
+	Val     int64
+}
+
+// JobByInstance helper struct to match job + instance ID against prom metrics
+type JobByInstance struct {
+	ID       string
+	Instance string
+}
+
+// NewFluxTest deploys AggregatorsNum flux aggregators concurrently
+func NewFluxTest(spec *FluxTestSpec) (*FluxTest, error) {
+	s, err := suite.DefaultLocalSetup(spec.InitFunc)
+	if err != nil {
+		return nil, err
+	}
+	clNodes, nodeAddrs, err := suite.ConnectToTemplateNodes()
+	if err != nil {
+		return nil, err
+	}
+	err = suite.FundTemplateNodes(s.Client, s.Wallets, clNodes, 9e18, 0)
+	if err != nil {
+		return nil, err
+	}
+	s.Client.(*client.EthereumClient).BorrowedNonces(true)
+	adapter, err := tools.NewExternalAdapter()
+	if err != nil {
+		return nil, err
+	}
+
+	fluxInstances := make([]contracts.FluxAggregator, 0)
+	nodesByHostPort := make(map[string]client.Chainlink)
+	contractToJobsMap := make(map[string][]JobByInstance)
+	mu := &sync.Mutex{}
+	g := &errgroup.Group{}
+	for i := 0; i < spec.AggregatorsNum; i++ {
+		deployFluxInstance(&FluxInstanceDeployment{
```

Comment:
Commented this before, but the goroutine call should be here, not wrapping the whole function implementation in a goroutine.